h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -328,11 +328,14 @@ private void parallelGridSearch(final Grid<MP> grid) {
 
     if(!startModels.isEmpty()) {
       parallelModelBuilder.run(startModels);
-      parallelModelBuilder.join();
+      parallelModelBuilder.join(); // Warning: keep in mind this is not being executed as a F/J task, it just looks like one
     }
     grid.update(_job);
 
     attemptGridSave(grid);
+    if (_job.stop_requested()) {
```

Comment:
at this point the grid is done, what is the point of throwing this exception?

Reply:
it is not done, only the models that were executed before the user triggered the "cancel"